### DISCUSSION TITLE
DS18B20 - How to get full 12bit serial number

### DISCUSSION THREAD

**Discussion by jpier123** [arendst/Tasmota#19898] [Q&A] [answered]
Hi All,

I'm looking for some advice about the best aproach with getting DS18B20 sensor full 12bit (16 character, example: 0x93955c4b09646128 ESPEasy: 28-61-64-09-4b-5c-95-93 ) serial number, which is necessery for  Alias name implementation. Any ideas? I know how to do it using ESPHome or ESPEasy, but would like try with Tasmota instead.

Thanks in Advance,

JP

**Comment by sfromis** [comment, marked as answer]
If you have a Tasmota build including support for the `DS18Alias` command, entering it without arguments will list existing sensors with the 64-bit `Id` value you can use when assigning an alias.

**Reply by jpier123** [reply]
Hi sfromis,

Yes Indeed! 

**Before:**
_CMD: DS18Alias
RSL: RESULT = {"DS18B20-1":{"Id":"B37FF0FF0264FF28"},"DS18B20-2":{"Id":"1B37387B0E64FF28"}}_

**After with Alias:**
_CMD: DS18Alias
RSL: RESULT = {"T7":{"Id":"B37FF0FF0264FF28"},"T8":{"Id":"1B37387B0E64FF28"}}_

Once more time - Thank You very much. I'm learning every day.


Best Regards,

JP
